docker-hub/publish/insights-analytics.md

```diff
@@ -0,0 +1,76 @@
+|Type | The industry from which the event originates. Industry types include “business”, “ISP” (internet service provider), “hosting”, “education”, and “null” in cases where the industry could not be identified.|
+|Host |The cloud service provider used in an event.|
+|Country |The country from which the request originated.  |
+|Timestamp |Date & time of an event in the following schema: YYYY-MM-DD 00:00:00|
+|Namespace |The Docker organization that a repository is a part of. |
+|Repo |The repository that an image belongs to.|
+|Reference |The tag or digest of any given image.|
+|HTTP Method | The HTTP method(s) used in a request by the client. More information on Docker Registry HTTP API protocols can be found here.|
+|User-Agent Tool|The application a user used to pull an image.  Extracted from the UA string.|
+|Unique IP Address|As part of our privacy-preserving policy, Docker only shares the count of distinct unique IP addresses that request an image.|
+
+
+
+## Action Classification Rule Set
+
+|Starting Event | Reference | Followed By | Resulting Action | Use Case(s) |Notes|
+|:-----|:--------|:-----|:--------|:-----|:--------|
+| HEAD | tag | N/A | Version Check | User already has all layers existing on local machine|This is similar to the use case of a pull by tag when the user already has all the image layers existing locally, however, we are able to differentiate the user intent and classify accordingly.|
+| Get| tag | N/A |Pull by Tag | User already has all layers existing on local machine and / or Image is single arch |
+| Get | tag | Get by different digest | Pull by tag | Image is multiarch |Following GET by digests must be different from the first digest|
+| HEAD | tag | GET by same digest | Pull by Tag | Image is multi-arch but some part of the image already exists on the local machine and/or User already has all layers existing on local machine|The HEAD by tag will send the most current digest, the following GET must be by that same digest. There may be an additional GET, if the image is multi arch (use case on the next line). If the user doesn't want the most current digest, then the user would perform HEAD by digest.|
+| HEAD| tag |GET by same digest, THEN a second GET by different digest| Pull by Tag| Image is multi-arch |The HEAD by tag will send the most current digest, the following GET must be by that same digest. Since the image is multi arch, there is a second GET by a different digest. If the user doesn't want the most current digest, then the user would perform HEAD by digest.|
+| HEAD | tag | GET by same digest, THEN a second GET by different digest | Pull by Tag| Image is multi arch | The HEAD by tag will send the most current digest, the following GET must be by that same digest. Since the image is multi-arch, there is a second GET by a different digest. If the user doesn't want the most current digest, then the user would perform HEAD by digest.|
```

Comment:
```suggestion
| HEAD | tag | GET by same digest, then a second GET by different digest | Pull by tag| Image is multi-arch | The HEAD by tag will send the most current digest, the following GET must be by that same digest. Since the image is multi-arch, there is a second GET by a different digest. If the user doesn't want the most recent digest, then the user would perform HEAD by digest.|
```